ktor-features/ktor-jrpc/src/io/ktor/jrpc/routers/JrpcRouter.kt

```diff
@@ -0,0 +1,191 @@
+package io.ktor.jrpc.routers
+
+import com.google.gson.JsonParseException
+import io.ktor.application.ApplicationCall
+import io.ktor.application.call
+import io.ktor.jrpc.JrpcSupport
+import io.ktor.jrpc.model.*
+import io.ktor.pipeline.PipelineContext
+import io.ktor.request.contentCharset
+import io.ktor.request.receiveOrNull
+import io.ktor.response.respond
+import io.ktor.util.pipeline.ContextDsl
+import kotlinx.coroutines.experimental.io.jvm.javaio.toInputStream
+import java.io.PrintWriter
+import java.io.StringWriter
+
+/**
+ * Type of HashMap<String, JrpcRouteItem<*>>
+ */
+typealias JrpcMethodRouter = HashMap<String, JrpcRouteItem<*>>
+
+/**
+ * JSON-RPC router class
+ * Use GSON from [JrpcSupport] for converting JRPC body
+ * Handle JRPC-Requests using configured handlers
+ * Before JRPC and After JRPC hooks can be set
+ */
+open class JrpcRouter {
```

Comment:
Why is it open?

Reply:
For constructions like this 

    jrpc("/json-rpc", MainRouter(environment.config))

and the MainRouter is defined in other file and use some internal logic (lacal variables, functions, etc.). In this case, kotlin file with routing will more readable without lists of methods implementations.
Or this is not a Ktor-way?

Reply:
Well, you can always extract your builder function to a separate file

Reply:
`jrpc("/json-rpc", ::mainRouter)` or `jrpc("/json-rpc") { mainRouter() }`

Reply:
That's right, so you recommend to change JrpcRouter to default (final) class?

Reply:
Yes, exactly. Users can extract function/class anyway so there is no need to make router open for inheritance. Also it could be internal (? not sure)